packages/amplify-graphql-api-construct/src/amplify-graphql-definition.ts

```diff
@@ -1,32 +1,90 @@
 import * as os from 'os';
 import { SchemaFile } from 'aws-cdk-lib/aws-appsync';
-import { IAmplifyGraphqlDefinition } from './types';
+import { IAmplifyGraphqlDefinition, ModelDataSourceDefinition } from './types';
+import { getModelTypeNames } from './internal';
 
+export const DEFAULT_MODEL_DATA_SOURCE_DEFINITION: ModelDataSourceDefinition = {
+  name: 'defaultDDB',
+  strategy: {
+    dbType: 'DYNAMODB',
+    provisionStrategy: 'DEFAULT',
+  },
+};
 /**
  * Class exposing utilities to produce IAmplifyGraphqlDefinition objects given various inputs.
  */
 export class AmplifyGraphqlDefinition {
   /**
    * Produce a schema definition from a string input
    * @param schema the graphql input as a string
+   * @param modelDataSourceDefinition the provision definition for `@model` datasource. The DynamoDB from CloudFormation will be used by default.
    * @returns a fully formed amplify graphql definition
    */
-  static fromString(schema: string): IAmplifyGraphqlDefinition {
+  static fromString(
+    schema: string,
+    modelDataSourceDefinition: ModelDataSourceDefinition = DEFAULT_MODEL_DATA_SOURCE_DEFINITION,
+  ): IAmplifyGraphqlDefinition {
     return {
       schema,
       functionSlots: [],
+      dataSourceProvisionConfig: {
+        project: modelDataSourceDefinition,
+      },
     };
   }
 
   /**
    * Convert one or more appsync SchemaFile objects into an Amplify Graphql Schema
    * @param filePaths one or more paths to the graphql files to process
+   * @param modelDataSourceDefinition the provision definition for `@model` datasource. The DynamoDB from CloudFormation will be used by default.
    * @returns a fully formed amplify graphql definition
    */
-  static fromFiles(...filePaths: string[]): IAmplifyGraphqlDefinition {
+  static fromFiles(
+    filePaths: string | string[],
+    modelDataSourceDefinition: ModelDataSourceDefinition = DEFAULT_MODEL_DATA_SOURCE_DEFINITION,
+  ): IAmplifyGraphqlDefinition {
+    if (!Array.isArray(filePaths)) {
+      filePaths = [filePaths];
+    }
     return {
       schema: filePaths.map((filePath) => new SchemaFile({ filePath }).definition).join(os.EOL),
       functionSlots: [],
+      dataSourceProvisionConfig: {
+        project: modelDataSourceDefinition,
+      },
+    };
+  }
+
+  /**
+   * Combines multiple IAmplifyGraphqlDefinitions into a single definition.
+   * @param definitions the definitions to combine
```

Comment:
We should document the error conditions. Off the top of my head, the only one I can think of is conflicting model definitions (if we find the same model in multiple definitions). For SQL support, we'll also disallow cross-type many-to-many relationships.

Reply:
I think this is better to address after the RDS changes are merged. Mark this as TODO